src/common/TokenParser.ts

```diff
@@ -1,43 +1,42 @@
 import * as winston from "winston";
-
 import { ERC20Contract } from "../models/Erc20ContractModel";
 import { Token } from "../models/TokenModel";
 import { Config } from "./Config";
 import { getTokenBalanceForAddress, loadContractABIs } from "./Utils";
 import { TransactionOperation } from "../models/TransactionOperationModel";
 import { NotParsableContracts } from "../models/NotParsableContractModel";
 
-
 export class TokenParser {
-
-    private abiList = loadContractABIs();
+    
+    private abi = require("./contracts/Erc20Abi");
     private abiDecoder = require("abi-decoder");
+    private OperationTypes = {
+        Transfer: "Transfer",
+    }
 
     constructor() {
-        for (const abi of this.abiList) {
-            this.abiDecoder.addABI(abi);
-        }
+        this.abiDecoder.addABI(this.abi);
     }
 
     public parseERC20Contracts(transactions: any) {
-        if (!transactions) {
-            return Promise.resolve([undefined, undefined]);
-        }
+        if (!transactions) return Promise.resolve([undefined, undefined]);
+
+        let contractAddresses: string[] = [];
 
-        // extract  valid contracts
-        let contractAddresses: any = [];
         transactions.map((transaction: any) => {
-            const decodedInput = this.abiDecoder.decodeMethod(transaction.input);
-            if (
-                (decodedInput && decodedInput.name === "transfer" && Array.isArray(decodedInput.params) && decodedInput.params.length === 2 && transaction.to !== null) ||
-                (decodedInput && decodedInput.name === "mint" && Array.isArray(decodedInput.params) && decodedInput.params.length === 2 && transaction.to !== null)
-            ) {
-                contractAddresses.push(transaction.to.toLowerCase());
+            const decodedLogs = this.abiDecoder.decodeLogs(transaction.receipt.logs);
+            if (decodedLogs.length > 0) {
+                decodedLogs.forEach((log: any) => {
+                    if (log) {
+                        if (log.name === this.OperationTypes.Transfer) {
+                            contractAddresses.push(transaction.to.toLowerCase());
+                        }
+                    }
+                });
             }
         });
-        // remove duplicates
-        contractAddresses = contractAddresses.filter((elem: any, pos: any, arr: any) => arr.indexOf(elem) == pos);
-        // process contracts
+
+        contractAddresses = contractAddresses.filter((elem: any, pos: any, arr: any) => arr.indexOf(elem) == pos);;
```

Comment:
Can you use `Set` in here to keep unique contracts?

Reply:
Fixed